### PR TITLE
Security improvements

### PR DESCRIPTION
**Pull request by vanmaegima**
## Readiness

* [ ] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

File: source/reference-manual/security/ff-security-summary.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+---------------------------
+
+.. list-table:: Secure OTA Certificates Summary
+   :header-rows: 1
+
+   * - Keys
+     - Type
+     - Owner
+   * - TUF Root Key
+     - RSA 4096
```
Comment (vanmaegima) [inline source/reference-manual/security/ff-security-summary.rst:81]: @vkhoroz have the tuf keys type changed already?

**Comment by doanac** [comment]
Docs for 365f4e4a9d0945318b2739a6c3b1628d4a12b60d are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1991/docs/artifacts/html/index.html

File: source/reference-manual/security/ff-security-summary.rst (hunk 1 above)
Comment (vkhoroz) [reply source/reference-manual/security/ff-security-summary.rst:81]: @vanmaegima yes.

- For new factories they are Ed25519 by default; but the customer may ask for RSA when factory is created.
- For the majority of old factories they are RSA.
- In either case the customer can switch from one key type to the other one whenever they want.

I don't know how to say that in one sentence, there should be an easy way.

File: source/reference-manual/security/ff-security-summary.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
+FoundriesFactory Security Summary
+=================================
+
+This page brings a brief summary of the many security aspects involved in a FoundriesFactory.
```
Comment (kprosise) [inline source/reference-manual/security/ff-security-summary.rst:4]:
```suggestion
This page provides a brief summary of the many security aspects involved in FoundriesFactory®.
```

File: source/reference-manual/security/ff-security-summary.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+   * - Keys
+     - Type
+     - Owner
+   * - Root of Trust key (*factory_ca.key*)
+     - EC prime256v1
+     - Owned and managed by the customer (offline key)
+   * - TLS key
+     - EC prime256v1
+     - Owned and managed by Foundries.IO (used for mTLS handshake)
```
Comment (kprosise) [inline source/reference-manual/security/ff-security-summary.rst:21]:
```suggestion
     - Owned and managed by Foundries.io (used for mTLS handshake)
```

File: source/reference-manual/security/ff-security-summary.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+   * - TUF Root Key
+     - RSA 4096
+     - Owned and managed by the customer (offline key)
+   * - TUF Snapshot Key
+     - RSA 4096
+     - Owned and managed by FoundriesFactory CI
+   * - TUF Timestamp Key
+     - RSA 4096
+     - Owned and managed by FoundriesFactory CI
+   * - TUF Targets Key - Online Target Signing Key
```
Comment (kprosise) [inline source/reference-manual/security/ff-security-summary.rst:89]:
```suggestion
   * - TUF Targets Key—Online Target Signing Key
```

File: source/reference-manual/security/ff-security-summary.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+     - RSA 4096
+     - Owned and managed by FoundriesFactory CI
+   * - TUF Targets Key - Customer Offline Signing Key
```
Comment (kprosise) [inline source/reference-manual/security/ff-security-summary.rst:92]:
```suggestion
   * - TUF Targets Key—Customer Offline Signing Key
```

File: source/reference-manual/security/secure-machines.rst [hunk 6]
```diff
@@ -118,3 +118,25 @@ to sign the SPL images, to fuse, and close the board.
   caution and only after understanding the critical implication of those commands.
 
 .. include:: imx-generic-custom-keys.rst
+
+Accessing Secure Storage
+------------------------
+
+Once a device has been successfully fused and closed, the secure storage (RPMB) becomes available.
+This is accessed through ``fiovb`` (Foundries.io Verified Boot) early trusted application from OP-TEE.
```
Comment (kprosise) [inline source/reference-manual/security/secure-machines.rst:126]:
```suggestion
This is accessed through ``fiovb`` (Foundries.io™ Verified Boot) early trusted application from Open Portable-Trusted Execution Environment (OP-TEE).
```

File: source/reference-manual/security/secure-machines.rst [hunk 7]
```diff
@@ -118,3 +118,25 @@ to sign the SPL images, to fuse, and close the board.
   caution and only after understanding the critical implication of those commands.
 
 .. include:: imx-generic-custom-keys.rst
+
+Accessing Secure Storage
+------------------------
+
+Once a device has been successfully fused and closed, the secure storage (RPMB) becomes available.
```
Comment (kprosise) [inline source/reference-manual/security/secure-machines.rst:125]:
```suggestion
Once a device has been successfully fused and closed, the secure storage—Replay Protected Memory Block (RPMB)—becomes available.
```

File: source/reference-manual/security/secure-machines.rst [hunk 8]
```diff
@@ -118,3 +118,25 @@ to sign the SPL images, to fuse, and close the board.
...
+This is accessed through ``fiovb`` (Foundries.io Verified Boot) early trusted application from OP-TEE.
+
+By default, the secure storage only holds the variables used by ``aktualizr-lite`` to handle the updates, which are previously stored in ``uboot-env`` for non-fused boards. A user can extend this to store custom variables that need to be made secure, like mac addresses, serial numbers and other critical device information.
```
Comment (kprosise) [inline source/reference-manual/security/secure-machines.rst:128]:
```suggestion
By default, the secure storage only holds the variables used by ``aktualizr-lite`` to handle the updates, previously stored in ``uboot-env`` for non-fused boards. You can extend this to store custom variables that need to be made secure, like mac addresses, serial numbers and other critical device information.
```

File: source/reference-manual/security/security.rst [hunk 9]
```diff
@@ -77,10 +80,20 @@ More information around Secure Boot aspects supported by LmP can be found in:
...
    tee-on-versal-acap
-   factory-keys
 
 See how to implement the `Secure Boot Firmware Updates`_ below.
 
+Secure Online Keys for Boot Stack
+---------------------------------
+
+FoundriesFactory uses online keys to sign the components from the boot stack during build time.
+More information on how these keys are used and how to modify them can be found in the page below.
```
Comment (kprosise) [inline source/reference-manual/security/security.rst:90]:
```suggestion
More information on how these keys are used and how to modify them can be found below.
```

**Review by kprosise — COMMENTED** [review]
suggestions made, mostly addressing the findings of the linter.

File: source/reference-manual/security/ff-security-summary.rst [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+     - Depends on the SoC
+     - Owned and managed by the customer (offline key)
+
+The Hardware Root of Trust depends on the SoC used. Please refer to :ref:`ref-secure-boot` pages and to the vendor reference manual for more information.
+
+Secure Online Keys for Boot Stack
+---------------------------------
+
+.. list-table:: LmP Build Certificates Summary
+   :header-rows: 1
```
Comment (ricardosalveti) [inline source/reference-manual/security/ff-security-summary.rst:48]: We should probably say that this is hardware dependent, as it depends on what is available as part of the boot stack.

e.g. on x86 we use UEFI secure boot instead, and the only one used here would be for kernel modules.

File: source/reference-manual/security/ff-security-summary.rst (hunk 1 above)
Comment (vkhoroz) [reply source/reference-manual/security/ff-security-summary.rst:81]: I'd say `Ed25519 (default) or RSA 4096` here, and add smth like `Can be selected at factory creation or changed later` on the next line.

File: source/reference-manual/security/ff-security-summary.rst (hunk 4 above)
Comment (vkhoroz) [reply source/reference-manual/security/ff-security-summary.rst:89]: This is too wordy IMHO. Can it be simply `Online TUF Targets Signing Key`?
I'd prepend `Offline|Online` to every TUF key we have.

File: source/reference-manual/security/ff-security-summary.rst [hunk 11]
```diff
@@ -0,0 +1,98 @@
+FoundriesFactory Security Summary
```
Comment (vkhoroz) [inline source/reference-manual/security/ff-security-summary.rst:1]: It is strange to see two `Security Summary` pages. Can this be renamed to something more specific e.g. `List of Crypto Keys Used by FoundriesFactory`? I don't see any information beyond that here.

File: source/reference-manual/security/security.rst [hunk 12]
```diff
@@ -135,3 +148,16 @@ Hardware Secure Module (Secure Element) specifics of select hardware platforms a
...
    secure-elements/secure-element.050
    secure-elements/se050-enablement
+
+
+.. _ref-security-summary:
+
+FoundriesFactory Security Summary
+---------------------------------
+
+A brief summary of the certificates used in a FoundriesFactory can be found in:
```
Comment (vkhoroz) [inline source/reference-manual/security/security.rst:158]: IIUC this page lists _crypto keys_, not _certificates_.
Also, I don't see any information except for the keys in it.

So, I believe it can be combined into one section with e.g. `factory-keys` under some common name. At least both pages seem to talk about keys; just `factory-keys` has more details about a signle sub-topic.

Smth like below should work well:
```
Crypto Keys Used by FoundriesFactory
------------------------------------------

A FoundriesFactory uses different cryptographic keys for various parts of the solution.
A full list of these keys can be found below:

.. toctree::
   :maxdepth: 1

   ff-security-summary   (Expand to "Summary of Crypto Keys Used by FoundriesFactory")

Below you can find more details about the cryptographic keys used to sign components from the boot stack during the build time:

.. toctree::
   :maxdepth: 1

   factory-keys   (Expand to "Crypto Keys Used by FoundriesFactory for Secure Boot at Build Time")
```

Does it make sense to you?

**Comment by vanmaegima** [comment]
Thanks for the reviews! My plan is to tackle everything tomorrow and send a new version.

File: source/reference-manual/security/security.rst (hunk 12 above)
Comment (vanmaegima) [reply source/reference-manual/security/security.rst:158]: I'm not sure I agree with this. The whole idea behind this change is to have a single easy to find page that brings all the crypto keys information together, so I would prefer to keep the summary page in its own section so it's clearer for the readers.
I will rename the pages though.

File: source/reference-manual/security/secure-machines.rst [hunk 13]
```diff
@@ -118,3 +118,28 @@ to sign the SPL images, to fuse, and close the board.
   caution and only after understanding the critical implication of those commands.
 
 .. include:: imx-generic-custom-keys.rst
+
+Accessing Secure Storage
+------------------------
+
+.. note::
+   The LmP leverages the eMMC Replay Protected Memory Block (RPMB) as secure storage. This section is only applicable for devices that provide this feature.
```
Comment (vanmaegima) [inline source/reference-manual/security/secure-machines.rst:126]: @ricardosalveti I added a note, is this ok?

File: source/reference-manual/security/factory-keys.rst [hunk 14]
```diff
@@ -1,7 +1,7 @@
 .. _ref-factory-keys:
 
-The LmP keys and certificates used during the build of a FoundriesFactory
-=========================================================================
+Crypto Keys Used by FoundriesFactory at Build Time
+==================================================
```
Comment (vanmaegima) [inline source/reference-manual/security/factory-keys.rst:5]: @vkhoroz I renamed this page per your suggestion

File: source/reference-manual/security/ff-security-summary.rst [hunk 15]
```diff
@@ -0,0 +1,104 @@
+Summary of Crypto Keys Used by FoundriesFactory
+===============================================
```
Comment (vanmaegima) [inline source/reference-manual/security/ff-security-summary.rst:2]: @vkhoroz renamed page

**Comment by doanac** [comment]
Docs for ebe061b5a79b1405d7f36923e962c80b95d80d38 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2002/docs/artifacts/html/index.html

File: source/reference-manual/security/ff-security-summary.rst [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+-----------------------------------
+
+.. list-table:: Device Gateway Certificates Summary
+   :header-rows: 1
+
+   * - Keys
+     - Type
+     - Owner
+   * - Root of Trust key (*factory_ca.key*)
+     - EC prime256v1
```
[the remark this answers is not in the document]
Comment (ricardosalveti) [reply source/reference-manual/security/ff-security-summary.rst:17]: prime256v1 = NIST P-256, either is fine, but usually when talking about the keytype from the standard pov, NIST P-256 is preferred.

File: source/reference-manual/security/ff-security-summary.rst [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+   * - Hardware Root of Trust Key
+     - Depends on the SoC
+     - Owned and managed by the customer (offline key)
+
+The Hardware Root of Trust depends on the SoC used. Please refer to :ref:`ref-secure-boot` pages and to the vendor reference manual for more information.
+
+Secure Online Keys for Boot Stack
+---------------------------------
+
+The exact list of keys used for the boot stack depends on the hardware used. Some platforms will not make use of all keys. A list of available keys for a LmP build can be found below:
```
[the remark this answers is not in the document]
Comment (ricardosalveti) [reply source/reference-manual/security/ff-security-summary.rst:47]: yup, thanks!

File: source/reference-manual/security/secure-machines.rst (hunk 13 above)
Comment (ricardosalveti) [reply source/reference-manual/security/secure-machines.rst:126]: yup, thanks!

File: source/reference-manual/security/ff-security-summary.rst (hunk 17 above)
Comment (kprosise) [inline source/reference-manual/security/ff-security-summary.rst:47]:
```suggestion
The exact list of keys used for the boot stack depends on the hardware used. Some platforms will not make use of all keys. A list of available keys for an LmP build can be found below:
```

**Review by kprosise — APPROVED** [review]
Just one little suggestion left, otherwise it LGTM

File: source/reference-manual/security/ff-security-summary.rst (hunk 16 above)
[the remark this answers is not in the document]
Comment (vkhoroz) [reply source/reference-manual/security/ff-security-summary.rst:17]: @vanmaegima interesting question. I guess @ricardosalveti's proposal of _NIST P-256_ is a good middle ground.

File: source/reference-manual/security/ff-security-summary.rst (hunk 17 above)
Comment (vkhoroz) [reply source/reference-manual/security/ff-security-summary.rst:47]: Actually, thanks @kprosise I was always wondering should I write `a HSM` or `an HSM`. That's not a question when talking, but when writing I was always on the fence. Now I know the answer.

File: source/reference-manual/security/ff-security-summary.rst [hunk 18]
```diff
@@ -0,0 +1,104 @@
+Summary of Crypto Keys Used by FoundriesFactory
+===============================================
+
+This page provides a brief summary of the many security aspects involved in FoundriesFactory®.
+For detailed information on each key, please check the relevant page under :ref:`ref-security`.
```
Comment (vkhoroz) [inline source/reference-manual/security/ff-security-summary.rst:5]:
```suggestion
This page provides a brief summary of the cryptographic keys used by FoundriesFactory®.
For detailed information on each key, please check the relevant page under :ref:`ref-security`.
```

File: source/reference-manual/security/ff-security-summary.rst (hunk 16 above)
[the remark this answers is not in the document]
Comment (vanmaegima) [reply source/reference-manual/security/ff-security-summary.rst]: Moved to NIST P-256 but kept prime256v1 where it's used as a command.

File: source/reference-manual/security/secure-machines.rst [hunk 19]
```diff
@@ -5,7 +5,7 @@
 Machines with secure aspects enabled by FoundriesFactory
 ========================================================
 
-LmP provides machines with secure aspects enabled by default when using
+The Linux microPlatform™ provides machines with secure aspects enabled by default when using
```
Comment (vkhoroz) [inline source/reference-manual/security/secure-machines.rst:8]: I see the text below uses LmP abbreviation. Shouldn't we mention it here first like below?

```suggestion
The Linux microPlatform™ (LmP) provides machines with secure aspects enabled by default when using
```

**Comment by doanac** [comment]
Docs for 34d85b02be62e95ee36d9381f778622e004bd361 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2007/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 6b1a4f7a30d1150faaa72c14b7905d2a9908978e are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2008/docs/artifacts/html/index.html

File: source/reference-manual/security/security.rst [hunk 20]
```diff
@@ -136,3 +149,15 @@ Hardware Secure Module (Secure Element) specifics of select hardware platforms a
    secure-elements/secure-element.050
    secure-elements/se050-enablement
    secure-elements/secure-element.tpm
+
+.. _ref-security-summary:
+
+FoundriesFactory Security Summary
+---------------------------------
+
+A brief summary of the crypto keys used in a FoundriesFactory can be found in:
```
Comment (vkhoroz) [inline source/reference-manual/security/security.rst:158]: Maybe this is just my personal taste preference... I just looks weird to me to see the section named `FoundriesFactory Security Summary` on a page which summarizes all the aspects of the FoundriesFactory security. The text says about crypto keys, so the section name needs to do that as well. Just IMHO.

File: source/reference-manual/security/security.rst [hunk 21]
```diff
@@ -18,6 +18,9 @@ Below sections focus on the following aspects:
   - how to securely store secrets on your devices;
 
 
+.. note::
+   For a brief summary of the security around FoundriesFactory, please check :ref:`ref-security-summary`.
```
Comment (vkhoroz) [inline source/reference-manual/security/security.rst:22]: I was thinking, maybe we can put a reference to that summary page here, like this:

```
A brief summary of the crypto keys used in a FoundriesFactory can be found in:

.. toctree::
   :maxdepth: 1

   ff-security-summary
```

I mean, reference that page straight in the Overview section, rather than appending it to the bottom.
What do you think?

**Review by vkhoroz — APPROVED** [review]
Looks good to me.

I only have a few minor suggestions.

File: source/reference-manual/security/security.rst (hunk 12 above)
Comment (vkhoroz) [reply source/reference-manual/security/security.rst]: @vanmaegima The way you did it makes sense to me. Thanks!

File: source/reference-manual/security/device-gateway.rst [hunk 22]
```diff
@@ -19,7 +19,7 @@ Root of trust - factory_ca.key / factory_ca.pem
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The PKI root of trust for your factory. You own the private key
-(EC prime256v1 by default). The corresponding x509 certificate is shared
+(NIST P-256 by default). The corresponding x509 certificate is shared
```
Comment (vkhoroz) [inline source/reference-manual/security/device-gateway.rst:22]: @vanmaegima I'm sorry for this vale error. I think, maybe, just P-256 would work here, as we already have the keys summary page which uses the full name.

**Comment by doanac** [comment]
Docs for c8457ec4d73d7b9cf8b4d0acf8f93ee04398b1b1 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2011/docs/artifacts/html/index.html

File: source/reference-manual/security/security.rst (hunk 20 above)
Comment (vkhoroz) [reply source/reference-manual/security/security.rst]: This new text you propose looks better.

The thing is: I care not only about text on this page, but also the navigation panel.
For example, this is how `cloud services` or `secure elements` expand now:
TOC expanded - good

And here is how these two new sections look like:
TOC expanded - weird

A visual problem I see is that the upper-level section repeats the majority of words in the lower-level section.
They simply ask for less text in that structure, or less structure, or fit more items into that structure.

Comment (vkhoroz) [reply source/reference-manual/security/security.rst]: I know this is quite a challenge to make this thing visually appealing.
If it were easy I'd just make a _"to this or that"_ kind of recommendation.
But, I clearly see the problem, while the solution seems to be more illusive.

Comment (vkhoroz) [reply source/reference-manual/security/security.rst]: @vanmaegima this is just one of the examples how this could be sorted out:
TOC-expanded-ok

It is not yet ideal, I believe you'd find better words for it, and potentially a better structure.
The thing is: I care less about the exact text/structure of the _Security page_, and more about the ease of navigation through the _Security panel_.

Comment (vkhoroz) [reply source/reference-manual/security/security.rst]: Just FYI: as an engineer I rarely read the page content until I find the necessary title on the navigation panel.
If the navigation is not intuitive - I'd rather search index or ask Google than read through all the intermediate pages.

Comment (vanmaegima) [reply source/reference-manual/security/security.rst]: Thanks for the inputs @vkhoroz , now I understand your point. I will review this again later this week and see what I can do. @kprosise any suggestions on this are welcome as well.

Comment (vanmaegima) [reply source/reference-manual/security/security.rst]: I agree with simplifying the words but I still don't know if I agree with grouping the two sections, it doesn't make much sense to me (in terms of keeping the pages sectioned and easy to find). I will take a closer look once I have the time and check again though.

Comment (kprosise) [reply source/reference-manual/security/security.rst]: I would then just simplify the wording then. We can always revisit it later. Perhaps it is best to just get it out there, and then make any adjustments based on feedback/observations from customers if needed.

**Comment by kprosise** [comment]
@vanmaegima we may go ahead and merge this whenever you're ready!
